homeassistant/components/climate/spider.py

```diff
@@ -0,0 +1,151 @@
+"""
+Support for Spider thermostats.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.spider/
+"""
+
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ATTR_TEMPERATURE, PLATFORM_SCHEMA, STATE_COOL, STATE_HEAT, STATE_IDLE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, TEMP_CELSIUS
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['itho_daalderop_api==1.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Spider thermostat."""
+    from itho_daalderop_api import IthoDaalderop_API, UnauthorizedException
+
+    username = config.get(CONF_USERNAME)
```

Comment:
Use `dict[key]` for required keys in the config schema.

Reply:
Done. Moved to hub/component